projects/table-builder/src/lib/classes/table-store.ts

```diff
@@ -103,13 +109,32 @@ export class TableStore extends ComponentStore<TableState> {
 
 
   setUserDefinedWidth = this.updater((state,colWidths:{key: string, widthInPixel:number}[]) => {
-    const userDefinedWidths = {...state.userDefinedWidths};
+    const userDefinedWidths = {...state.userDefined.widths};
     colWidths.forEach(cw => {
       userDefinedWidths[cw.key] = cw.widthInPixel;
     });
-    return {...state, userDefinedWidths: userDefinedWidths};
+    return {...state, userDefined: {...state.userDefined,widths:userDefinedWidths}};
   });
 
+  setUserDefinedOrder = this.updater((state,moved:{key:string,newOrder:number})=>{
+    const userDefinedOrder = {...state.userDefined.order};
+
+    const newOrder = moved.newOrder;
+    const oldOrder = state.metaData[moved.key]._internalOrder;
+    const mdsArr = this.orderViewableMetaData(state.metaData);
+    moveItemInArray(mdsArr,oldOrder,newOrder);
+    const mds = {...state.metaData};
+    mdsArr.forEach((md,index) => {
+      mds[md.key]._internalOrder = index;
+
+      if(userDefinedOrder[md.key] != undefined){
```

Comment:
Would it be simpler if you just saved all of them?

Reply:
Nafks mina if the **code** changes to move around some columns. I only wanted to preserve the column the user actually moved to keep it in the same place.

Reply:
I agree, but I'm not sure it's worth it.
Because it's not at all obvious if the user meant column t2 or before column 3 or after column 1.
Therefore, I don't believe there's a great way to decide what is the correct way to handle code changes.

Hence, let's go for simplicity.

Reply:
Since you agree I would prefer to keep as is. Being as we don't know the intent we will just preserve what the action not the intention. I don't see that the other way would be simpler, it would just remove the `if` I think.

Reply:
I agree that it matters for code change.
I still disagree that this is better than the alternative due to the added complexity.

Reply:
Hence, let's go for simplicity.